data/README.md

```diff
@@ -80,3 +88,30 @@ const unsubscribe = wp.data.subscribe( () => {
 // Unsubcribe.
 unsubscribe();
 ```
+
+### `wp.data.onSubscribe( mapDataToProps: function )( WrappedComponent: Component )`
+
+If you use a React or WordPress Element, a Higher Order Component is made available to inject data into your components like so:
+
+```js
+const Component = ( { title } ) => <div>{ title }</div>;
+
+wp.data.onSubscribe( () => {
+	return {
+		title: wp.data.select( 'myPlugin' ).getTitle(),
+	};
+} )( Component );
+```
+
+You can also use this Higher Order Component to provide action handlers to your components
+
+```js
+const Component = ( { title, updateTitle } ) => <input value={ title } onChange={ updateTitle } />;
+
+wp.data.onSubscribe( () => {
+	return {
+		title: select( 'myPlugin' ).getTitle(),
```

Comment:
Now, you have to use `wp.data.select` and `wp.data.dispatch` in here. I liked the previous version more where you had it exposed as the part of `mapSelectToProps`. Can we also expose `wp.data` as a first param here?

```js
wp.data.onSubscribe( ( { dispatch, select } ) => {
	return {
		title: select( 'myPlugin' ).getTitle(),
		updateTitle: dispatch( 'myPlugin' ).setTitle,
	};
} )( Component );
```